### PR TITLE
HPCC-15570 Get WUQueryDetails by queryID or query alias

### PR DESCRIPTION
**Pull request by wangkx**
The existing WsWorkunits.WUQueryDetails can only retrieve
query details by queryID.

Signed-off-by: wangkx kevin.wang@lexisnexis.com

### CODE REVIEW

**Comment by wangkx** [comment]
@afishbeck please review. BTW: I cannot find out the function which returns a query tree based on queryID or alias. I also need the queryRegistry after I have the query tree.

**Comment by afishbeck** [comment]
@wangkx You only need to call resolveQueryAlias().  There is a version that takes the registry as a parameter since you need the registry after.

If you look at the closely at the implementation, it first checks if the string is a aueryalias, if it does it uses that, if not it uses the given string as the queryid.

File: esp/scm/ws_workunits.ecm [hunk 1]
```diff
@@ -1434,7 +1434,8 @@ ESPresponse [exceptions_inline] WUQueryFilesResponse
 
 ESPrequest WUQueryDetailsRequest
 {
-    string QueryId;
+    [depr_ver("1.61")] string QueryId;
+    [min_ver("1.61")] string QueryIdOrAlias;
```
Comment (afishbeck) [inline esp/scm/ws_workunits.ecm]: Don't need this, just one parameter.

File: esp/services/ws_workunits/ws_workunitsQuerySets.cpp [hunk 2]
```diff
@@ -1588,26 +1588,33 @@ bool CWsWorkunitsEx::onWUQueryFiles(IEspContext &context, IEspWUQueryFilesReques
...
-    PROGLOG("WUQueryDetails: QuerySet %s, query %s", querySet, queryId);
 
+    bool foundAlias = false;
+    const char* queryId = queryIdOrAlias;
     Owned<IPropertyTree> queryRegistry = getQueryRegistry(querySet, false);
-
-    StringBuffer xpath;
-    xpath.clear().append("Query[@id='").append(queryId).append("']");
-    IPropertyTree *query = queryRegistry->queryPropTree(xpath.str());
+    Owned<IPropertyTree> query = getQueryById(queryRegistry, queryId);
```
Comment (afishbeck) [inline esp/services/ws_workunits/ws_workunitsQuerySets.cpp]: Don't call this...

File: esp/services/ws_workunits/ws_workunitsQuerySets.cpp [hunk 3]
```diff
@@ -1588,26 +1588,33 @@ bool CWsWorkunitsEx::onWUQueryFiles(IEspContext &context, IEspWUQueryFilesReques
...
     if (!query)
     {
-        DBGLOG("No matching Query");
-        throw MakeStringException(ECLWATCH_QUERYID_NOT_FOUND,"No matching query for query ID %s.", queryId);
+        query.setown(resolveQueryAlias(queryRegistry, queryIdOrAlias));
```
Comment (afishbeck) [inline esp/services/ws_workunits/ws_workunitsQuerySets.cpp]: ...only call this.

File: esp/services/ws_workunits/ws_workunitsQuerySets.cpp (hunk 2 above)
Comment (wangkx) [inline esp/services/ws_workunits/ws_workunitsQuerySets.cpp]: Thanks for the help. I will change this.

File: esp/scm/ws_workunits.ecm (hunk 1 above)
Comment (wangkx) [inline esp/scm/ws_workunits.ecm]: The version may be needed if a client wants to know which request parameter can be used for a specific version of the client.

Comment (afishbeck) [inline esp/scm/ws_workunits.ecm]: @wangkx No, that kind of confuses interface version with implementation version.  I think keeping it one parameter but adding support for aliases in the background is ok, at least in this case.

Comment (wangkx) [inline esp/scm/ws_workunits.ecm]: Ok. I will remove new QueryIdOrAlias.

**Comment by wangkx** [comment]
@afishbeck please review again.

**Review by afishbeck — APPROVED** [review]
@wangkx looks good.

**Comment by wangkx** [comment]
@richardkchapman this PR is ready to be merged.

**Pull request merged**